File: Recursion and Backtracking/RecursionInArray/maxEleArray.cpp
#include<iostream>
#include<vector>

using namespace std;

int maxEleArray(vector <int> & arr, int idx){
    if(idx==arr.size()) return -1e9;

    return max(arr[idx], maxEleArray(arr, idx+1));
}

int main(){
    int n; cin>>n;
    vector<int>arr(n,0);
    for(int i=0; i<n; i++) cin>>arr[i];

    cout<<maxEleArray(arr,0)<<endl;
    return 0;    
}

File: Recursion and Backtracking/RecursionInArray/maxEleArray.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws Exception {
        Scanner scn = new Scanner(System.in);
        int n = scn.nextInt();
        int[] arr = new int[n];
        for (int i = 0; i < n; i++)
            arr[i] = scn.nextInt();
        System.out.print(maxOfArray(arr, 0));
    }

    public static int maxOfArray(int[] arr, int idx) {
        if(idx==arr.length) return -(int)1e9;
        
        return Math.max(arr[idx], maxOfArray(arr, idx+1));
    }

}